libs/domain/cart/src/services/resolver/cart.resolver.ts

```diff
@@ -14,11 +14,13 @@ export type CartResolvers = {
 };
 
 export class CartResolver extends BaseResolver<CartResolvers> {
-  protected cartService$ = resolve(CartService);
+  constructor(protected cartService = inject(CartService)) {
+    super();
+  }
```

Comment:
BC:
- existing protected `cartService$` no longer exists
- existing class extensions would fail since this class has a mandatory constructor

Fix:
- make cartService optional, keep `cartService$`, but deprecate it and support both during 1.0
- optionally, try to use feature flag to bring changes in